spec/features/admin/questionnaire_spec.rb

```diff
@@ -15,66 +15,38 @@
     it "works" do
       # create
       visit admin_loan_tab_path(loan, 'questions')
-      fill_in('response_set[summary[text]]', with: 'kittens jumping on rainbows')
-      click_button 'Save Responses'
-      expect(page).to have_content 'successfully created'
-      expect(page).to have_content 'kittens jumping on rainbows'
-      expect(criteria.summary.text).to eq 'kittens jumping on rainbows'
 
-      # edit
-      click_link('Edit Responses')
-      fill_in('response_set[summary[text]]', with: 'sexy unicorns')
-      click_button 'Save Responses'
-      expect(page).to have_content 'successfully updated'
-      expect(page).to have_content 'sexy unicorns'
-      expect(page).not_to have_content 'Warning'
-      expect(criteria.summary.text).to eq 'sexy unicorns'
-
-      # delete
-      accept_confirm { click_link('Delete All Responses') }
-      expect(page).to have_content 'successfully deleted'
-      expect(page).not_to have_content 'sexy unicorns'
-      # After deletion, should be in edit mode
-      expect(page).not_to have_content 'Edit Responses'
-      expect(page).to have_selector 'input[value="Save Responses"]'
+      # check that edit-all button turns on edit mode
+      expect(page).not_to have_content("Now editing")
+      page.find(".edit-all", match: :first).click
```

Comment:
I think you can write `first(".edit-all").click` to achieve the same thing, which is rather less verbose. Have you tried that? If not can you see if it works?

If it does, same simplifications throughout please.

Reply:
cool, thanks for this pointer